src/simulator/human_object.cpp

```diff
@@ -0,0 +1,128 @@
+//========================================================================
+//  This software is free: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License Version 3,
+//  as published by the Free Software Foundation.
+//
+//  This software is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  Version 3 in the file COPYING that came with this distribution.
+//  If not, see <http://www.gnu.org/licenses/>.
+//========================================================================
+/*!
+  \file    human_object.cpp
+  \brief   C++ Interface: Human Object
+  \author  Yifeng Zhu, (C) 2020
+  \email   yifeng.zhu@utexas.edu
+*/
+//========================================================================
+
+#include "human_object.h"
+
+HumanObject::HumanObject(){
+  // x, y, angle
+  pose_ = Eigen::Vector3f(0., 0., 0.);
+  // just a cylinder for now
+  double r = 0.3;
+  int num_segments = 20;
+
+  double angle_increment = 2 * M_PI / num_segments;
+
+  Eigen::Vector2f v0(r, 0.);
+  Eigen::Vector2f v1;
+  double eps = 0.001;
+  for (int i = 1; i < num_segments; i++){
+    v1 = Eigen::Rotation2Df(angle_increment * i) * Eigen::Vector2f(r, 0.0);
+
+    // TODO: Fix the vector map bug that closed shape would have wrong occlusion
+    Eigen::Vector2f eps_vec = (v1 - v0).normalized() * eps;
+    template_lines_.push_back(geometry::line2f(v0 + eps_vec, v1 - eps_vec));
+    v0 = v1;
+  }
+  template_lines_.push_back(geometry::line2f(v1, Eigen::Vector2f(r, 0.0)));
+  pose_lines_ = template_lines_;
+  this->initialize();
+}
+
+HumanObject::HumanObject(std::string config_file){
+  // TODO: Load the shape, start point, goal point and walking mode from a config file
+  this->initialize();
+}
+
+
+HumanObject::~HumanObject(){
+
+}
+
+void HumanObject::initialize(){
+  start_pose_ = pose_;
+  present_ = true;
```

Comment:
This doesn't appear to be used?

Reply:
This function is called in the constructor. I was thinking to have all the same intiailization from different constructors to be in this function. Not sure if there is better way to do this?

Reply:
I know the function is. What does `present_` do?

Reply:
Oh you mean the present_ ? Probably I'll just delete that for now